sympy/stats/crv.py

```diff
@@ -171,13 +336,35 @@ def __new__(cls, *args):
     def check(*args):
         pass
 
-    def sample(self, size=()):
+    def sample(self, size=1, library='python'):
         """ A random realization from the distribution """
-        icdf = self._inverse_cdf_expression()
-        if not size:
-            return icdf(random.uniform(0, 1))
+
+        if library == 'python':
+            samps = getattr(SampleExternal, '_sample_python')(self, size)
+            if samps != None:
+                return list(samps) if size!=1 else samps[0]
+            else:
+                icdf = self._inverse_cdf_expression()
+                if size == 1:
+                    return icdf(random.uniform(0, 1))
+                else:
+                    return [icdf(random.uniform(0, 1))]*size
```

Comment:
```suggestion
```

Reply:
IMO,` 'python' `needs to be handled differently. Because it has an alternative if `SampleExternal` fails for `'python'`

Reply:
Why are using that alternative? Can that alternative be shifted to `_sample_python` so that final result is not `None` for `_sample_python` whatever be the case.